streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -393,39 +393,41 @@ private void recordSendError(final String topic,
                 "indicating the task may be migrated out";
             sendException.set(new TaskMigratedException(errorMessage, productionException));
         } else {
-            if (isRetriable(productionException)) {
+            final DefaultErrorHandlerContext errorHandlerContext = new DefaultErrorHandlerContext(
+                null, // only required to pass for DeserializationExceptionHandler
+                context.recordContext().topic(),
```

Comment:
Do we need a check that the `recordContext()` itself is not `null` first?

Reply:
Guess you are right -> https://issues.apache.org/jira/browse/KAFKA-17527 (working on a PR for a fix already, and can update this PR later for it...)

Reply:
Other PR was merged. Resolved via rebasing.